packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx

```diff
@@ -2,156 +2,107 @@ import React, { FC, useState } from 'react';
 import { ItemRowFragment } from '@openmsupply-client/system';
 import {
   BasicSpinner,
-  useDialog,
   Divider,
-  TableContainer,
-  TabContext,
-  TabList,
-  Tab,
   useTranslation,
   useIsMediumScreen,
-  ButtonWithIcon,
-  PlusCircleIcon,
   Box,
   ModalMode,
   useNotification,
-  DialogButton,
 } from '@openmsupply-client/common';
-import { BatchTable, PricingTable } from './StocktakeLineEditTables';
-import { StocktakeLinePanel } from './StocktakeLinePanel';
 import { StocktakeLineEditForm } from './StocktakeLineEditForm';
 import { useStocktakeLineEdit } from './hooks';
-
+import {
+  StocktakeLineEditTabs,
+  StyledTabContainer,
+  StyledTabPanel,
+} from './StocktakeLineEditTabs';
+import { BatchTable, PricingTable } from './StocktakeLineEditTables';
+import { StocktakeLineEditModal } from './StocktakeLineEditModal';
 interface StocktakeLineEditProps {
   item: ItemRowFragment | null;
   mode: ModalMode | null;
   onClose: () => void;
   isOpen: boolean;
 }
 
-enum Tabs {
-  Batch = 'Batch',
-  Pricing = 'Pricing',
-}
-
 export const StocktakeLineEdit: FC<StocktakeLineEditProps> = ({
   item,
   mode,
   onClose,
   isOpen,
 }) => {
   const { error } = useNotification();
-  const { Modal } = useDialog({ onClose, isOpen });
   const [currentItem, setCurrentItem] = useState(item);
-  const [currentTab, setCurrentTab] = useState(Tabs.Batch);
   const isMediumScreen = useIsMediumScreen();
   const t = useTranslation(['common', 'inventory']);
-
   const { draftLines, update, addLine, isLoading, save, nextItem } =
     useStocktakeLineEdit(currentItem);
 
-  return (
-    <Modal
-      title={
-        mode === ModalMode.Create
-          ? t('heading.add-item')
-          : t('heading.edit-item')
-      }
-      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
-      nextButton={
-        <DialogButton
-          variant="next"
-          onClick={async () => {
-            try {
-              await save(draftLines);
-              if (nextItem) setCurrentItem(nextItem);
-              else onClose();
+  const onNext = async () => {
+    await save(draftLines);
+    if (nextItem) setCurrentItem(nextItem);
+    else onClose();
```

Comment:
I rather like it - but the behaviour is different to all the other 'Ok & Next's which just stop at the end.